src/main/java/com/google/devtools/build/lib/syntax/StringModule.java

```diff
@@ -909,9 +916,21 @@ public Integer count(String self, String sub, Integer start, Object end)
             defaultValue = "None",
             doc = "optional position at which to stop comparing.")
       })
-  public Boolean endsWith(String self, String sub, Integer start, Object end)
+  public Boolean endsWith(String self, Object sub, Integer start, Object end)
       throws ConversionException {
-    return pythonSubstring(self, start, end, "'end' operand of 'endswith'").endsWith(sub);
+    String str = pythonSubstring(self, start, end, "'end' operand of 'endswith'");
+    if (sub instanceof String) {
+      return str.endsWith((String)sub);
+    }
+
+    @SuppressWarnings("unchecked")
+    Tuple<String> subs = (Tuple<String>) sub;
+    for (String s : subs) {
```

Comment:
I believe we don't check the element type, that the `generic1` part of `ParamType` is just for documentation generation purposes. This should be replaced by

    for (String s : subs.getContents(String.class, "string")) {
        ...
    }

Same for `startsWith` below.

Reply:
Could be a good idea to document that.

Reply:
Indeed, I'll add that caveat to Param's javadoc now.

Note that the var `subs` and the `@SuppressWarnings` are no longer needed, since the cast is done by `getContents`. `ConversionException` extends `EvalException` so this can just be `EvalException`. (Same below, of course.)